backend/runner/taskcheck/statement_affected_rows_report_executor.go

```diff
@@ -312,42 +294,81 @@ func reportStatementAffectedRowsForMySQL(ctx context.Context, sqlDB *sql.DB, sta
+		if node, ok := node.(*tidbast.InsertStmt); ok && node.Select == nil {
+			return int64(len(node.Lists)), nil
+		}
+		if dbType == db.OceanBase {
+			return getAffectedRowsCount(ctx, sqlDB, fmt.Sprintf("EXPLAIN FORMAT=JSON %s", node.Text()), getAffectedRowsCountForOceanBase)
+		}
+		return getAffectedRowsCount(ctx, sqlDB, fmt.Sprintf("EXPLAIN %s", node.Text()), getAffectedRowsCountForMysql)
 	default:
 		return 0, nil
 	}
 }
 
-func getInsertAffectedRowsForMysql(ctx context.Context, sqlDB *sql.DB, node *tidbast.InsertStmt) (int64, error) {
-	if node.Select == nil {
-		return int64(len(node.Lists)), nil
-	}
-	res, err := query(ctx, sqlDB, fmt.Sprintf("EXPLAIN %s", node.Text()))
+// OceanBaseQueryPlan represents the query plan of OceanBase.
+type OceanBaseQueryPlan struct {
+	ID       int    `json:"ID"`
+	Operator string `json:"OPERATOR"`
+	Name     string `json:"NAME"`
+	EstRows  int64  `json:"EST.ROWS"`
+	Cost     int    `json:"COST"`
+	OutPut   any    `json:"output"`
+}
+
+// Unmarshal parses data and stores the result to current OceanBaseQueryPlan.
+func (plan *OceanBaseQueryPlan) Unmarshal(data any) error {
+	b, err := json.Marshal(data)
 	if err != nil {
-		return 0, err
+		return err
 	}
-	rowCount, err := getAffectedRowsCountForMysql(res)
-	if err != nil {
-		return 0, errors.Wrapf(err, "failed to get insert affected rows count for mysql, res %+v", res)
+	if b != nil {
+		return json.Unmarshal(b, &plan)
 	}
-	return rowCount, nil
+	return nil
 }
 
-func getUpdateOrDeleteAffectedRowsForMysql(ctx context.Context, sqlDB *sql.DB, node tidbast.StmtNode) (int64, error) {
-	res, err := query(ctx, sqlDB, fmt.Sprintf("EXPLAIN %s", node.Text()))
-	if err != nil {
-		return 0, err
+func getAffectedRowsCountForOceanBase(res []any) (int64, error) {
+	// the res struct is []any{columnName, columnTable, rowDataList}
+	if len(res) != 3 {
+		return 0, errors.Errorf("expected 3 but got %d", len(res))
 	}
-	rowCount, err := getAffectedRowsCountForMysql(res)
-	if err != nil {
-		return 0, errors.Wrapf(err, "failed to get update or delete affected rows count for mysql, res %+v", res)
+	rowList, ok := res[2].([]any)
+	if !ok {
+		return 0, errors.Errorf("expected []any but got %t", res[2])
 	}
-	return rowCount, nil
+	if len(rowList) < 1 {
+		return 0, errors.Errorf("not found any data")
+	}
+
+	plan, ok := rowList[0].([]any)
+	if !ok {
+		return 0, errors.Errorf("expected []any but got %t", rowList[0])
+	}
+	planString, ok := plan[0].(string)
+	if !ok {
+		return 0, errors.Errorf("expected string but got %t", plan[0])
+	}
+	var planValue map[string]any
+	_ = json.Unmarshal([]byte(planString), &planValue)
+	if len(planValue) > 0 {
+		queryPlan := OceanBaseQueryPlan{}
+		_ = queryPlan.Unmarshal(planValue)
+		if queryPlan.Operator != "" {
+			return queryPlan.EstRows, nil
+		}
+		for _, v := range planValue {
+			child := OceanBaseQueryPlan{}
+			_ = child.Unmarshal(v)
+			if child.Operator != "" {
+				return child.EstRows, nil
+			}
+		}
+	}
```

Comment:
Let's not ignore errors.

Reply:
The for range loop here is used to get the fields with prefix 'CHILD_', and there will be errors when parsing other plain text fields. I think it's parts of the try case, we can leave it before a better solution.
```json
{
    "ID":3,
    "OPERATOR":"EXCHANGE IN",
    "NAME":"EXCHANGE IN",
    "EST.ROWS":6,
    "COST":44,
    "output":"nil",
    "CHILD_1":{
        "ID":2,
        "OPERATOR":"EXCHANGE OUT",
        "NAME":"EXCHANGE OUT",
        "EST.ROWS":6,
        "COST":42,
        "output":"nil",
        "CHILD_1":{
            "ID":1,
            "OPERATOR":"DELETE",
            "NAME":"DELETE",
            "EST.ROWS":6,
            "COST":38,
            "output":"byt",
            "CHILD_1":{
                "ID":0,
                "OPERATOR":"TABLE SCAN",
                "NAME":"TABLE SCAN",
                "EST.ROWS":6,
                "COST":2,
                "output":[
                    "bytebase.id",
                    "bytebase.name"
                ]
            }
        }
    }
}
```

Reply:
Perhaps we can use map[string]json.RawMessage?
https://stackoverflow.com/questions/11066946/partly-json-unmarshal-into-a-map-in-go

Reply:
Done.